pkg/graph/resolvers/apply_changes.go

```diff
@@ -11,20 +12,45 @@ import (
 // Code largely copied from GQLGen's docs on changesets
 // https://gqlgen.com/reference/changesets/
 func ApplyChanges(changes map[string]interface{}, to interface{}) error {
+	// Perform some pre-processing of the changes
+	// This is done here rather than in DecodeHook because it's easy to infer intent on things like "" -> nil or [] -> nil here
```

Comment:
When you use block comments to describe a chunk of code, there's a good chance that block could be extracted to a named method to simplify/clarify the code